platform/mbed_application.c

```diff
@@ -118,12 +125,8 @@ __asm static void start_new_application(void *sp, void *pc)
 void start_new_application(void *sp, void *pc)
 {
     __asm volatile (
-#if defined(__CORTEX_M23)
-        "movw   r2, #0      \n"
+        "movw   r2, #0      \n" // Fail to compile "mov r2, #0" with ARMC6. Replace with movw/movt.
         "movt   r2, #0      \n"
```

Comment:
Sorry - my previous comments weren't very clear!  MOVT is superfluous here as MOVW will zero-extend.  So we can remove the MOVT instruction.  "MOVW r2, #0" will put 0x0 into all 32-bits of r2.

Reply:
@TacoGrandeTX Yes, **MOVT** is superfluous. I remove it in https://github.com/ARMmbed/mbed-os/pull/6949/commits/23dcd82fabcb81a40078fc904d604d63997aca6c.